### ISSUE TITLE
Ubuntu 16.04 Xenial End of Standard Support

### ISSUE THREAD

**Issue by oerdnj** [oerdnj/deb.sury.org#1567]
In April 2021, Ubuntu 16.04 Xenial will reach [End of Standard Support](https://wiki.ubuntu.com/Releases) and will be available only as a paid option through [Ubuntu Extended Security Maintenance](https://ubuntu.com/security/esm).

What does it mean for DEB.SURY.ORG PPAs?

* The packages for Ubuntu 16.04 will be deleted shortly after the EoL/EoSS is announced, usually at the same time as the next PHP release is published because it's not possible to build the packages any more.
* The packages for Ubuntu 16.04 will be available via [PHP LTS by Freexian](https://php.freexian.com) paid program. This is cheaper option than previously announced [Private dedicated repositories](https://www.patreon.com/posts/private-ubuntu-30596045).

With the experience of Ubuntu 14.04 Trusty EOL, I can't stress enough that you should plan ahead and have your servers already upgraded by now. Yesterday was too late.

**Comment by augustedupin** [comment]
Until April 2021? but right now, i tried to install php7.1 in ubuntu 16.04 LTS and the package is not found. maybe I'm missing something, but i tried everything to install it but this statement is my only clue.

**Comment by oerdnj** [comment]
@augustedupin the PHP 7.1 packages are available in the PPA and I am going to delete your comment and mine just in a moment. This is not meant to provide user support.

**Comment by andrewfenn** [comment]
I understand the need to monetize the service you're offering. It is a good service. I would ask you to reconsider deleting packages from the mirror though. By deleting the deb files in the mirror you're breaking systems and creating a situation where it's too risky to even migrate to your paid LTS service as there is no way to rollback. You will also catch a lot of people off guard who just followed some tutorial using your mirror from many years ago. They don't realize they're one apt-get install away from breaking their webserver with no way to go back. They most likely never saw your warnings before like myself wondering why I can not install the [zip module](http://ppa.launchpad.net/ondrej/php/ubuntu/pool/main/p/php7.2-zip_7.2.14-1+ubuntu14.04.1+deb.sury.org+1_amd64.deb) for PHP anymore. In launchpad you can keep old versions in the mirror without having to build or support them any longer. I would ask you to think about doing this in the future.

**Comment by oerdnj** [comment]
No. And I am not having this discussion all over again every two years.

Those are the conditions - I am not providing packages for distributions past EOL date. Take it or leave it.

**Comment by andrewfenn** [comment]
Just for myself and others benefit as I am sure there will be many people coming to this with the same problem... do you still have copies of the old files for sale or is the paid support completely different deb packages? Is it possible to purchase the old deb files I need?

**Comment by oerdnj** [comment]
Ubuntu 16.04 files haven't been removed yet.  There was no PHP release since Xenial end-of-life.

Ubuntu 14.04 file you are requesting has been removed 2 years ago.

The packages will be different (and up-to-date) because of smaller set of libraries backported. Also there would be no such service, the goal here isn't to monetize the service, but to provide packages with all the security patches applied.

I haven't changed my stance since the inception of the service. I am not going to provide access to outdated vulnerable packages.

**Comment by cpotor** [comment]
oerdnj  is just a full of him and ignorant person that gets satisfaction harming others , but is not his fault we had broken systems ,the ones who advertised on tutorials for new linux users about using that ppa to install software without to tell also it can be deleted have a fault too . unless somebody can provide a mirror of what was on his crazy bastard ppa before he to delete xenial files i keep having disgrace about all he do and i find him a misserable person he just gets happy twiter there are users who no longer have access to files

**Comment by dianaxxyyzz** [comment]
oerdnj is just a full of him and ignorant person that gets satisfaction harming others , but is not his fault we had broken systems ,the ones who advertised on tutorials for new linux users about using that ppa to install software without to tell also it can be deleted have a fault too . unless somebody can provide a mirror of what was on his crazy bastard ppa before he to delete xenial files i keep having disgrace about all he do and i find him a misserable person he just gets happy twiter there are users who no longer have access to files

**Comment by dianaxxyyzz** [comment]
oerdnj is just a full of him and ignorant person that gets satisfaction harming others , but is not his fault we had broken systems ,the ones who advertised on tutorials for new linux users about using that ppa to install software without to tell also it can be deleted have a fault too . unless somebody can provide a mirror of what was on the crazy bastard  ppa before he to delete xenial files i keep having disgrace about all he do and i find him a miserable person, he just gets happy on  twiter that there are users who no longer have access to files . f*ck you misearable and s*ck my d** . I hope you enjoy all tutorials have your shit ppa's in them and ppls used them without to tell you will delete files as soon as you have a chance to ask for money . i hope you will s*ck it well


and for the ones who dnt want or have  way  to pay the crazy bastard ,here are what you can use to have php5.6 and 7.4 without xampp or build by yourself
deb http://ppa.launchpad.net/hemanthkumar/xenialphp5.6/ubuntu xenial main
deb-src http://ppa.launchpad.net/hemanthkumar/xenialphp5.6/ubuntu xenial main


deb http://ppa.launchpad.net/jczaplicki/xenial-php74-temp/ubuntu xenial main
deb-src http://ppa.launchpad.net/jczaplicki/xenial-php74-temp/ubuntu xenial main

if they will delete this too I will put them on one of my private servers , we like to help not to be happy when ppls have problems like this bastard do , oh yeh you very smart you learned to compile php and now you want to be bilionare from this ,go sck it good

**Comment by dianaxxyyzz** [comment]
oerdnj is just a full of him and ignorant person that gets satisfaction harming others , but is not his fault we had broken systems ,the ones who advertised on tutorials for new linux users about using that ppa to install software without to tell also it can be deleted have a fault too . unless somebody can provide a mirror of what was on the crazy bastard  ppa before he to delete xenial files i keep having disgrace about all he do and i find him a miserable person, he just gets happy on  twiter that there are users who no longer have access to files . f*ck you misearable and s*ck my d** . I hope you enjoy all tutorials have your shit ppa's in them and ppls used them without to tell you will delete files as soon as you have a chance to ask for money . i hope you will s*ck it well


and for the ones who dnt want or have no  way  to pay the crazy bastard ,here are what you can use to have php5.6 and 7.4 without xampp or build by yourself
deb http://ppa.launchpad.net/hemanthkumar/xenialphp5.6/ubuntu xenial main
deb-src http://ppa.launchpad.net/hemanthkumar/xenialphp5.6/ubuntu xenial main


deb http://ppa.launchpad.net/jczaplicki/xenial-php74-temp/ubuntu xenial main
deb-src http://ppa.launchpad.net/jczaplicki/xenial-php74-temp/ubuntu xenial main

if they will delete this too I will put them on one of my private servers , we like to help not to be happy when ppls have problems like this bastard do , oh yeh you very smart you learned to compile php and now you want to be bilionare from this ,go sck it good

**Comment by robocoder** [comment]
@dianaxxyyzz Since I've received 4 email notifications from your rant, I'll remind you that Github's Terms of Service includes the following:

> That means you should be respectful and civil at all times, [...]

Reference: https://docs.github.com/en/github/site-policy/github-community-guidelines#what-is-not-allowed

**Comment by oerdnj** [comment]
Thanks @robocoder (also because the comments got deleted without a trace and your comment has the github username) and sorry that you all had to read this. I've blocked and reported the abusive user.

**Comment by andrewfenn** [comment]
@oerdnj Thanks for hard work you put in. Some of us might not agree on your specific policies you've made clear, but we all do appreciate your work. Please ignore the minority of negative comments like what was posted by that user.

**Comment by trantor** [comment]
@oerdnj First thanks a lot for the repos and packages still under maintenance. However, as far as the 16.04 packages are concerned, on the [PHP LTS by Freexian](https://php.freexian.com) page it seems as though they have not been published (yet?). Could you, by any chance, provide some more info on timing?

> * The packages for Ubuntu 16.04 will be available via [PHP LTS by Freexian](https://php.freexian.com) paid program. This is cheaper option than previously announced [Private dedicated repositories](https://www.patreon.com/posts/private-ubuntu-30596045).

**Comment by oerdnj** [comment]
Somebody has to fund the initial packaging and as far as I know this has just recently happened, but since I am mostly on vacation, I don't know any details. The initial set of packages are to be published by the end of August or early September.

**Comment by own3mall** [comment]
Does anyone have the last uploaded PHP 7.4 packages for Xenial that I can download from somewhere?  Or, does anyone have an Ubuntu 16.04 system with these packages already installed?  It would be possible to obtain the dpkg files from an Ubuntu 16.04 system already running the previously released PHP 7.4 packages.  I could then mirror these packages or create my own repository.  

Since the author isn't capable of coming up with valid reasons as to why users still run older no longer supported versions of Ubuntu (or any operating system for that manner), it would be nice to be able to obtain the last released packages from somewhere.  As of right now, I cannot find them anywhere on the internet, and that's wrong.  I don't need any support.  I just want to run PHP 7.4 on Ubuntu 16.04.  That's all.  I accept any and all risks associated with using that previously released software.  I don't understand why @oerdnj is essentially forcing us to run "supported" operating system versions for no valid reason other than that the old stuff is unsupported (which means pretty much next to nothing... nothing running on those systems is going to stop working).  I still have systems that run XP, and just because they're unsupported doesn't mean I can't run previously released software on them.

I don't need a baby sitter telling me how to run my systems when I'm perfectly capable of doing it myself.  

I will never understand this mentality that some developers have.  Just because something is newer doesn't make it better.  I get it though, you don't want to support old software, and that's perfectly fine, but could you at least make the last release obtainable?

**Comment by oerdnj** [comment]
> I don't understand why @oerdnj is essentially forcing us to run "supported" operating system versions for no valid reason

I don't force you do anything. These are the conditions under I provide the packages, nobody is preventing you from compiling the PHP from sources.

Now, please take you rant elsewhere it is not welcome here.

**Comment by own3mall** [comment]
You've created problems by not keeping the old packages available.  Whereas, no problems would have been created had you just kept them online.  

Which one is better?  Creating problems, or avoiding potential problems?  

It's not a rant, and my opinion is certainly welcome here.  It's sad that I'm now going to have to mirror and download every single package you create so I can make it available to others should they need it in the future when you decide to cut them off for no valid reason other than time has passed.

**Comment by oerdnj** [comment]
Let me be absolutely clear here - if you come here ranting, act all entitled and demand I do stuff for you for free, your are going to get blocked and you rants will be deleted. The word *free* in the free software doesn't mean you are entitled to command other people to do what you want or need. It's about sharing, helping to build better world and being nice to each other. If you think that free means that you have to fill your plate with free stuff and you yell at the staff that the buffet is out of crab cakes, you are not welcome here. This is about bringing your best food to have a nice picnic in the park.

**Comment by Remzi1993** [comment]
> Let me be absolutely clear here - if you come here ranting, act all entitled and demand I do stuff for you for free, your are going to get blocked and you rants will be deleted. The word _free_ in the free software doesn't mean you are entitled to command other people to do what you want or need. It's about sharing, helping to build better world and being nice to each other. If you think that free means that you have to fill your plate with free stuff and you yell at the staff that the buffet is out of crab cakes, you are not welcome here. This is about bringing your best food to have a nice picnic in the park.

I fully agree! I still don't understand why some people feel entitled to someone's time who is doing things for free. Also happened to me (I was a maintainer of something small). If it's that important I would advise them to hire us to make it happen, but that's not what they want. They want us to work for free and extra hard, can't have it both ways and cherry pick.

**Comment by axlotl** [comment]
"The word free in the free software doesn't mean you are entitled to command other people to do what you want or need."

No, but it does mean "free" as in no money. When I installed slackware in 1996 I *paid* for the little stack of 3.5" floppies. No prob. But the software on them was free.
